support/k8s/k8s-workload-registrar/controller.go

```diff
@@ -210,6 +220,51 @@ func (c *Controller) makeID(pathFmt string, pathArgs ...interface{}) string {
 	return id.String()
 }
 
+// Ensure that there is exactly one entry matching these selectors, and it's this one
```

Comment:
>since the registrar lives in the server pod

Probably also useful to note that this is not technically required... Registration API can be reached remotely, but then the registrar needs an SVID and some extra setup is required

Reply:
Oh yes, there is an UpdateEntry RPC. I'll rearrange things around that, and just spit out an error if there are duplicates, that's much better.

The idea of duplicate registrars generating noise is interesting... it's harmless beyond log spam, but we could extend the registration protocol to be explicit that we want to discard exact duplicates and consider that success.

For avoiding changing webhook behaviour: easy enough to guard this new behaviour behind a config flag, and have it default on for informer and off for webhook, does that resolve the issue? I can still split it out if needs be.

Reply:
Now reworked with UpdateEntry, that's much neater. (Waiting on feedback on the other two ideas before I do something with them)

Reply:
CreateEntryIfNotExists resolved the log spam case, so now we're just down to: what to do about the new behaviour for webhooks?